### DISCUSSION TITLE
[SquareLine Studio] ]Displaying Temperature and Pressure Values on CYD

### DISCUSSION THREAD

**Discussion by amirzohaib** [witnessmenow/ESP32-Cheap-Yellow-Display#234] [General]
Hi everyone,

I just began working on CYD and designed my own UI to display temp and pressure..etc 
I have made Text fields and i thought i could just enter Variables as text and itll show the values

I somehow can't understand, how can i pass Sensor values to those fields, or manipulate a Bar graph etc..
since I can't link a variable in Square Line Studio.

After googling, I can see It's something like this
lv_label_set_text_fmt(ui_Label1, “%d”, data);

but where to put it in my code? In the Main function?
And usually I had a delay in my loop so that i don't get bombarded with values, but now there is a ticker 5ms in the loop for display, im not sure how i could refresh sensor values every 2 seconds if that ticker needs a 5ms refresh in the main loop

Thanks for any help, i really am new to this, the price was just unbeatable (7,65EUR incl. shipping for a CYD2USB) so i want to use it everywhere

**Comment by Wires2005** [comment]
Hi amirzohaib,

I'm new to this also but a little further along the learning experience.

I write my values to the screen like this...

```
char pressure[10];
char temperature[10];
sprintf(pressure, "%.1f", bmp388.pressure);
sprintf(temperature, "%.1f", bmp388.temperature);
lv_label_set_text(ui_PressReading, pressure);
lv_label_set_text(ui_TempReading, temperature);
```

ui_PressReading & ui_TempReading are the names of the text field in SquareLine.

I place this code in it's own function but it's called from the main function every 250ms after I poll the sensor for current readings.

As for scheduling the writing of the values, you don't want to use delays. I'm using the Arduino IDE and I use millis() and an interval value.

**Reply by amirzohaib** [reply]
Yeah i managed to get it working, but im not sure how can i best implement that call every 250ms for eg. ive had just put the Label set in the loop

**Reply by Wires2005** [reply]
There are multiple ways to do timing/scheduling/multitasking.

For starters, have a look at the following.
https://forum.arduino.cc/t/using-millis-for-timing-a-beginners-guide/483573

In your case, instead of blinking an LED, you'll put your code in to refresh your readings and update the lvgl text labels.
